[[ SB Inclusions ]] Allow script library inclusions on mobile platforms

livecodeali opened the pull request:
Builds on https://github.com/livecode/livecode/pull/3999

Commit:
[[ SB Inclusions ]] Add helper function to translate inclusions into deploy params

Essentially this just abstracts what was already happening on desktop - take the
settings, and build a list of auxiliary stackfiles and a startup script to
initialise them.

It also resaves all the auxiliary stackfiles as non-script only stacks which
prevents a crash on startup.

Finally, it computes the selected lcb extension dependencies and sets the deploy
'modules' param accordingly.

Commit:
[[ SB Inclusions ]] Use the deploy params helper when building desktop platforms

Commit:
[[ SB Inclusions ]] Use the deploy params helper when building for mobile

peter-b commented on ide-support/revsblibrary.livecodescript:
```diff
@@ -1719,3 +1719,80 @@ command revSBResolveIncludedExternal pPlatform, @xFile
+   -- If the datagrid is used, then include the library as an aux stack
+   if "data grid templates" is in the subStacks of stack pStack and there is a stack "revDataGridLibrary" then
+      appendToStringList the effective filename of stack "revDataGridLibrary", tAuxiliaryStackFiles
+   end if
```
Is there any possibility that the new inclusions abstraction could be used to remove this special case in the future (not in this PR)?

livecodeali replied:
I suppose the way to think of it is as a custom control with a support library, so it should probably be something stated in a 'manifest' for the data grid control. Ideally we would handle it in the same way as any other custom control package once we have a package manager.

peter-b commented:
This all looks good subject to merging #3999 and adding a release note!

Commit:
Add release note

Commit:
Merge branch 'develop' of github.com:livecode/livecode into feature-full_mobile_inclusions

livecodeali commented:
@livecode-vulcan review by @peter-b ok 5684470

livecode-vulcan commented:
💙 review by @peter-b ok 568447069b84c60ddac4ac2ef2cb84232d46fd13

livecode-vulcan commented:
😎 test success 568447069b84c60ddac4ac2ef2cb84232d46fd13
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2497)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2507)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2490)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2508)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2514)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2518)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2530)

runrevmark commented:
This is cool - but we should probably be careful with regards libURL as it has slightly different semantics than the in-built networking stuff on iOS and Android. I _think_ the current implementation with the libURL engine messages means that libUrl will override the libUrl implementation... We should probably double check this before putting it into a release...

livecodeali commented:
@runrevmark Yes you're right - It only called the engine implementation if the getUrl (or equivalent) message is passed. What's the best way of dealing with that then? Make it not available on mobile? Or just warn if it is included?

livecodeali commented:
@livecode-vulcan review not ok 5684470

livecode-vulcan commented:
💔 review by @livecodeali not ok 568447069b84c60ddac4ac2ef2cb84232d46fd13

Commit:
Merge branch 'develop' of github.com:livecode/livecode into feature-full_mobile_inclusions

Commit:
[[ SB Inclusions ]] Confirm inclusions which override engine implementations

Also ignore inclusions with engine implementations where they were added by
the search for inclusions process.

Commit:
[[ SB Inclusions ]] Script libs are supported on mobile

livecodeali commented:
Requires https://github.com/livecode/livecode-ide/pull/1171

Commit:
[[ SB Inclusions ]] Resaving datagrid stack causes dg not to work on android

runrevmark commented:
@livecode-vulcan review ok 1c0d2a9

livecode-vulcan commented:
💙 review by @runrevmark ok 1c0d2a962ede8c3d70f6a4f06b7ae8ae9f2f3353

livecode-vulcan commented:
😎 test success 1c0d2a962ede8c3d70f6a4f06b7ae8ae9f2f3353
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2516)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2526)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2511)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2527)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2533)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2537)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2549)
